### PR TITLE
Memkind and Jemalloc static link

### PR DESCRIPTION
**Pull request by KFilipek**
Static linking for Jemalloc and Memkind to avoid problems with deps. Solved in the same way as liblua from deps.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/redis/63)

### CODE REVIEW

**Comment by michalbiesek** [comment]
I am not sure if You aware that redis-cli and redis-benchmark also used libjemalloc.so.2
ldd redis-cli
	libjemalloc.so.2 => not found
ldd redis-benchmark
	libjemalloc.so.2 => not found

File: src/Makefile [hunk 1]
```diff
@@ -125,9 +125,9 @@ ifeq ($(MALLOC),jemalloc)
 	FINAL_CFLAGS+= -DUSE_JEMALLOC -I$(JEMALLOC_PATH)/include
 #	FINAL_CFLAGS+= -I$(JEMALLOC_PATH)/jemalloc/obj/include
 	FINAL_CFLAGS+= -DJE_PREFIX=je_
-	FINAL_CFLAGS+= -I$(MEMKIND_PATH)/include
+    FINAL_CFLAGS+= -I$(MEMKIND_PATH)/include
```
Comment (michalbiesek) [inline src/Makefile]: Please remove this change ( be consistent here with tab )

File: src/Makefile [hunk 2]
```diff
@@ -202,7 +202,7 @@ endif
 
 # redis-server
 $(REDIS_SERVER_NAME): $(REDIS_SERVER_OBJ)
-	$(REDIS_LD) -o $@ $^ ../deps/hiredis/libhiredis.a ../deps/lua/src/liblua.a $(FINAL_LIBS)
+	$(REDIS_LD) -o $@ $^ ../deps/hiredis/libhiredis.a ../deps/lua/src/liblua.a ../deps/memkind/.libs/libmemkind.a ../deps/memkind/jemalloc/obj/lib/libjemalloc.a ../deps/jemalloc_JE/lib/libjemalloc.a $(FINAL_LIBS) -lnuma
```
Comment (michalbiesek) [inline src/Makefile]: Consider in linking static library usage of variables:
https://github.com/pmem/redis/blob/8856349a6319d148bff8ea2528a35328b85101e8/src/Makefile#L26-L27

File: src/Makefile (hunk 1 above)
Comment (KFilipek) [reply src/Makefile]: Done

File: src/Makefile (hunk 2 above)
Comment (KFilipek) [reply src/Makefile]: I wanted to be consistent with Redis style. I made change that you suggested.

File: src/memkind_malloc.c [hunk 3]
```diff
@@ -33,19 +34,22 @@ void *memkind_calloc_wrapper(size_t size) {
         size = jemk_malloc_usable_size(ptr);
         update_memkind_malloc_stat_alloc(size);
     }
+    printf("Calloc:\t%p\n",ptr);
```
Comment (michalbiesek) [inline src/memkind_malloc.c]: I think changes in memkind_malloc.c file are unrelated to  "Memkind and Jemalloc static link"

Comment (KFilipek) [reply src/memkind_malloc.c]: Why I pushed this, sorry

File: src/Makefile [hunk 4]
```diff
@@ -218,11 +215,11 @@ $(REDIS_CHECK_AOF_NAME): $(REDIS_SERVER_NAME)
...
 $(REDIS_CLI_NAME): $(REDIS_CLI_OBJ)
-	$(REDIS_LD) -o $@ $^ ../deps/hiredis/libhiredis.a ../deps/linenoise/linenoise.o $(FINAL_LIBS)
+	$(REDIS_LD) -o $@ $^ ../deps/hiredis/libhiredis.a $(JEMALLOC_PATH)/lib/libjemalloc.a ../deps/linenoise/linenoise.o $(FINAL_LIBS)
 
 # redis-benchmark
 $(REDIS_BENCHMARK_NAME): $(REDIS_BENCHMARK_OBJ)
-	$(REDIS_LD) -o $@ $^ ../deps/hiredis/libhiredis.a $(FINAL_LIBS)
+	$(REDIS_LD) -o $@ $^ ../deps/hiredis/libhiredis.a $(JEMALLOC_PATH)/lib/libjemalloc.a $(FINAL_LIBS)
 
 dict-benchmark: dict.c zmalloc.c sds.c siphash.c
```
Comment (michalbiesek) [inline src/Makefile]: I found that this change got impact for 2 other configuration:
-make dict-benchmark
-make test-sds
They also used jemalloc. Maybe it is good idea to add "$(JEMALLOC_PATH)/lib/libjemalloc.a" in FINAL_LIBS ?

Comment (KFilipek) [reply src/Makefile]: Good point, fixed

**Comment by michalbiesek** [comment]
Reviewed 2 of 4 files at r2, 1 of 1 files at r4.
Review status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @karpiop)

---



*Comments from [Reviewable](https://reviewable.io/reviews/pmem/redis/63)*

**Comment by michalbiesek** [comment]
:lgtm:

---

Review status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @karpiop)

---



*Comments from [Reviewable](https://reviewable.io/reviews/pmem/redis/63#-:-LEo6l_C38OLs6F6jcwl:bnfp4nl)*
